### ISSUE TITLE
Search records ignoring records where the column does not exist.

### ISSUE THREAD

**Issue by cold01water** [openobserve/openobserve#1007]
### Which OpenObserve functionalities are relevant/related to the feature request?

log search

### Description

message was divided into fields A, B, C, and D with the VRL function. Some logs do not match the rule and the field does not exist.

In this case, if you search by specifying the fields A, B, C, and D, the search cannot be performed. You will get the following error

```
ERROR openobserve::service::search::datafusion::exec] query sql execute failed, session: Session { id: "fcf0e4fa-e09e-40fc-8b74-5ef06768d0be-0", storage_type: FsMemory }, sql: SELECT action FROM tbl WHERE (_timestamp >= 1687410384049000 AND _timestamp < 1687411284049000)  LIMIT 150, err: SchemaError(FieldNotFound { field: Column { relation: None, name: "_timestamp" }, valid_fields: [Column { relation: Some(Bare { table: "tbl_temp" }), name: "action" }] })
```

If you try to format it in advance, you will have to use VRL, but currently you cannot assign the VRL function until after the stream is created.
And the stream is created when you save the log.
In other words, since there is no timing to allocate the function, noise is always generated.

I am having particular trouble creating a dashboard.

### Proposed solution

Noise in the logs is so frequent that it is difficult to format everything.
It is best to allow searches even when columns are missing.

Alternatively, it would be nice to be able to truncate noise in the search criteria or use a function such as is_exist() to exclude noise from the search.

### Alternatives considered

We considered pre-formatting with vector, etc., but it was difficult to remove all the noise.

**Comment by cold01water** [comment]
It seems the movement was not what I expected. I will close it for now.
